src/mmw/js/src/modeling/tests.js

```diff
@@ -376,43 +377,124 @@ describe('Modeling', function() {
+
+                    assert.equal(model.get('modification_hash'), modification_hash);
+                    assert.equal(model.get('inputmod_hash'), inputmod_hash);
+                });
+
                 it('generates and sets modification_hash based on the scenario\'s modifications', function() {
                     var model = new models.ScenarioModel({
                             modifications: [modificationsSample1]
                         });
 
                     model.updateModificationHash();
-                    assert.equal(model.get('modification_hash'), 'b0eb4af3250d9c1320d659805afaf049');
+                    assert.equal(model.get('modification_hash'), '40c4fdd89b06a0e9b7e1c3c5c2bd1f17');
 
                     var mod = new models.ModificationModel(modificationsSample2);
                     model.get('modifications').add(mod);
-                    assert.equal(model.get('modification_hash'), '07fb74f0e09eb1c31f3db8c78219758f');
+                    assert.equal(model.get('modification_hash'), 'dcda6a697d33a3bc95c95cc401e774bb');
 
                     model.get('modifications').remove(mod);
-                    assert.equal(model.get('modification_hash'), 'b0eb4af3250d9c1320d659805afaf049');
+                    assert.equal(model.get('modification_hash'), '40c4fdd89b06a0e9b7e1c3c5c2bd1f17');
                 });
 
                 it('is called when the modifications for a scenario changes', function() {
-                    var spy = sinon.spy(models.ScenarioModel.prototype, 'updateModificationHash'),
-                        model = new models.ScenarioModel({});
+                    var model = new models.ScenarioModel({});
 
                     model.get('modifications').add(new models.ModificationModel(modificationsSample1));
-                    assert.isTrue(spy.calledOnce);
-
-                    models.ScenarioModel.prototype.updateModificationHash.restore();
+                    assert.isTrue(modSpy.called);
                 });
 
                 it('is called before model#attemptSave when the modifications change', function() {
-                    var modSpy = sinon.spy(models.ScenarioModel.prototype, 'updateModificationHash'),
-                        saveSpy = sinon.spy(models.ScenarioModel.prototype, 'attemptSave'),
-                        model = new models.ScenarioModel({});
+                    var model = new models.ScenarioModel({});
 
                     model.addModification(new models.ModificationModel(modificationsSample1));
                     assert.isTrue(modSpy.calledBefore(saveSpy));
+                });
+            });
+
+            describe('#scenarioPolling', function() {
+                it('saves after polling finishes successfully', function(done) {
+                    var model = getTestScenarioModel(),
+                        tr55Response = '{"status":"started", "job":"8aef636e-2079-4f87-98dc-471d090141ad"}',
```

Comment:
We may want to move some of these huge json responses over to their own file for convenience. It might also encourage reuse.